### ISSUE TITLE
Switch manual to sphinx?

### ISSUE THREAD

**Issue by htgoebel** [pyinstaller/pyinstaller#1763]
The manual is already written in reST, so I wonder if we should change the format to sphinx.

Pro: This would allow to leverage the power of sphinx (if we need it ;-)
Con: We need to create our custom templates or layout to keep our look-and-feel.

Regarding our custom building of the documentation:
- `_definitions.txt` could be implemented with an `rst_epilog` in the `conf.py`. Some parts (e.g. version) are handled by sphinx already anyway.
- pdf version - either using [Latexbuilder](http://www.sphinx-doc.org/en/stable/_modules/sphinx/builders/latex.html#LaTeXBuilder) or `rst2pdf.pdfbuilder` (see the Latexbuilder link)
- man-pages can be build, too, see http://www.sphinx-doc.org/en/stable/_modules/sphinx/builders/manpage.html#ManualPageBuilder
- sphinx still uses a Makefile, so the could reuse everything we already have if we want or need to.

**Comment by nchammas** [comment]
Side question: As part of a potential switch to Sphinx, would we be interested in dropping the [PDF and HTML builds](https://github.com/pyinstaller/pyinstaller/tree/5427395dd8aa7ea558a23bd061f088ecd9f0a963/doc) of the docs from the repo?

They are essentially duplicate information since the RST is the source of truth, and I wonder if we can just do without them or have them be dynamically generated somehow.

**Comment by bjones1** [comment]
I think this is an excellent idea and have meant to work on this for some time, but haven't found the time to. I agree with @nchammas that using https://readthedocs.org/ would mean there's no need for HTML or PDF.

**Comment by matysek** [comment]
+1 for sphinx and dropping pdf.
Btw, do you know if anybody still needs pdf?

**Comment by htgoebel** [comment]
I'm going to work on this. Implementation plan:
- For testing I'll work on my personal repository https://github.com/htgoebel/pyinstaller
- Remove obsolete file
- Cut Manual into pieces
- Generate boiler plate for sphinx
- Local tests
- Integrate with readthedocs.
- Push to this repository.

**Comment by htgoebel** [comment]
Implemented in f6578abfd.

I used the short-url `rtfd.org` in the Readme, etc.

@pyinstaller/core-developers Still there are some topics to decide. You comments are very welcome:
1. Shall future versions of the documentation still reside at pythonhosted?
2. If yes: Shall pyinstaller.rtfd.org point to the stable release?

**Comment by matysek** [comment]
@htgoebel add 1: I don't think that's necessary. The sphinx doc looks great. Just updating the links to point at new location. I assume that sphinx and readthedoc allows better automation.

**Comment by htgoebel** [comment]
Sphinx for Pythonhosted could also be automated: https://pythonhosted.org/an_example_pypi_project/buildanduploadsphinx.html

**Comment by bjones1** [comment]
1. No. readthedocs is a great way to host and build Sphinx. Why bother with anything else?

Great job! This is a huge step forward in terms of providing more usable, readable docs. Well done.

**Comment by codewarrior0** [comment]
I agree with @bjones1: I think we should do away with pythonhosted and keep all the documentation on one site.

rtfd.org does allow you to have multiple versions of the documentation. I clicked the green "v: latest" at the bottom and got a menu of different versions (and output formats); choosing "stable" gave me [this unfinished site](http://pyinstaller.readthedocs.org/en/stable/).

**Comment by htgoebel** [comment]
> choosing "stable" gave me this unfinished site.

This is simply because we have no sphinx-build documentation for the last release.RTD used the tags to determine stable versions.So there is not even a chance to backport the docs to an existing release.

(Well, there could be a way, e.g. temporary changing the source-repo, add some minimal docu and a fake-tag there, and push that RTD, build stable there, revert to the original source repo. On the other hand, I wonder why the README is not rendered. I assume, this is due to our file `tests/functional/data/sphinx/conf.py`, which RTD takes as marker for the documentation base.)

**Comment by htgoebel** [comment]
> I assume, this is due to our file tests/functional/data/sphinx/conf.py, which RTD takes as marker for the documentation base.

I checked the build at RTD and indeed this is the case.

**Comment by htgoebel** [comment]
I consider this done. Since we decided to move the official documentation from pythonhosted to RTD with the next release, no further action is required on this ticket.

**Comment by nchammas** [comment]
@htgoebel - So to confirm, once the migration is complete:
* The docs will live [here on RTD](http://pyinstaller.readthedocs.org/en/stable/).
* The docs will no longer be hosted on PythonHosted.
* The PDF and HTML versions of the docs will be removed from git.

Is that correct?

**Comment by tallforasmurf** [comment]
When working on the doc I often had problems with the makefile and especially with PDF conversion, so I would not be sad to see it go.

In my simple use of sphinx to document another project, I found that sphinx's HTML output required quite a few files aside from the generated text. To include the doc in the bundle, I had to add these as data files when bundling. However that will not be an issue with docs hosted on RTD.

Should #1915 be merged into this issue?
